File: shop-app-web-user/src/main/web/src/app/models/productOrder.ts
import {Basket} from "./basket";

export class ProductOrder {
    private _id: number;
    private _ownerFirstName: string;
    private _ownerLastName: string;
    private _ownerPhone: string;
    private _ownerEmail: string;
    private _status: string;
    private _date: Date;
    private _basket: Basket;


    constructor(id: number, ownerFirstName: string, ownerLastName: string, ownerPhone: string, ownerEmail: string, status: string, date: Date, basket: Basket) {
        this._id = id;
        this._ownerFirstName = ownerFirstName;
        this._ownerLastName = ownerLastName;
        this._ownerPhone = ownerPhone;
        this._ownerEmail = ownerEmail;
        this._status = status;
        this._date = date;
        this._basket = basket;
    }

    get id(): number {
        return this._id;
    }

    set id(value: number) {
        this._id = value;
    }

    get ownerFirstName(): string {
        return this._ownerFirstName;
    }

    set ownerFirstName(value: string) {
        this._ownerFirstName = value;
    }

    get ownerLastName(): string {
        return this._ownerLastName;
    }

    set ownerLastName(value: string) {
        this._ownerLastName = value;
    }

    get ownerPhone(): string {
        return this._ownerPhone;
    }

    set ownerPhone(value: string) {
        this._ownerPhone = value;
    }

    get ownerEmail(): string {
        return this._ownerEmail;
    }

    set ownerEmail(value: string) {
        this._ownerEmail = value;
    }

    get status(): string {
        return this._status;
    }

    set status(value: string) {
        this._status = value;
    }

    get date(): Date {
        return this._date;
    }

    set date(value: Date) {
        this._date = value;
    }

    get basket(): Basket {
        return this._basket;
    }

    set basket(value: Basket) {
        this._basket = value;
    }
}

File: server/src/main/java/com/medochek/shopapp/domain/ProductOrder.java
package com.medochek.shopapp.domain;

import lombok.*;

import javax.persistence.*;
import java.time.LocalDateTime;

@Table
@Entity
@Data
@Builder
@AllArgsConstructor
@ToString(exclude = "basket")
public class ProductOrder {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String ownerFirstName;
    private String ownerLastName;
    private String ownerPhone;
    private String ownerEmail;
    private String status;
    private LocalDateTime date;
    private String priceList;

    @OneToOne
    @JoinColumn(name = "basket_id")
    private Basket basket;

    public ProductOrder(Long id, String ownerFirstName, String ownerLastName, String ownerPhone, String ownerEmail,
                        String status, LocalDateTime date, Basket basket) {
        this.id = id;
        this.ownerFirstName = ownerFirstName;
        this.ownerLastName = ownerLastName;
        this.ownerPhone = ownerPhone;
        this.ownerEmail = ownerEmail;
        this.status = status;
        this.date = date;
        this.basket = basket;
        this.priceList = "";
        for (BasketRow basketRow : basket.getBasketRowList()) {
            this.priceList += basketRow.getProduct().getId() + "-";
            if(basketRow.getProduct().getPriceSale() != null) {
                this.priceList += basketRow.getProduct().getPriceSale() + ";";
            }
            else {
                this.priceList += basketRow.getProduct().getPrice() + ";";
            }
        }
    }
}
